// @progress/kendo-react-excel-export/dist/npm/main.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KendoOoxml = exports.ExcelExportGroupHeader = exports.ExcelExportGroupFooter = exports.ExcelExportColumn = exports.ExcelExportFooter = exports.ExcelExportColumnGroup = exports.ExcelExport = void 0;
var ExcelExport_1 = require("./ExcelExport");
Object.defineProperty(exports, "ExcelExport", { enumerable: true, get: function () { return ExcelExport_1.ExcelExport; } });
var ExcelExportColumn_1 = require("./ExcelExportColumn");
Object.defineProperty(exports, "ExcelExportColumn", { enumerable: true, get: function () { return ExcelExportColumn_1.ExcelExportColumn; } });
var ExcelExportColumnGroup_1 = require("./ExcelExportColumnGroup");
Object.defineProperty(exports, "ExcelExportColumnGroup", { enumerable: true, get: function () { return ExcelExportColumnGroup_1.ExcelExportColumnGroup; } });
var ExcelExportFooter_1 = require("./templates/ExcelExportFooter");
Object.defineProperty(exports, "ExcelExportFooter", { enumerable: true, get: function () { return ExcelExportFooter_1.ExcelExportFooter; } });
var ExcelExportGroupFooter_1 = require("./templates/ExcelExportGroupFooter");
Object.defineProperty(exports, "ExcelExportGroupFooter", { enumerable: true, get: function () { return ExcelExportGroupFooter_1.ExcelExportGroupFooter; } });
var ExcelExportGroupHeader_1 = require("./templates/ExcelExportGroupHeader");
Object.defineProperty(exports, "ExcelExportGroupHeader", { enumerable: true, get: function () { return ExcelExportGroupHeader_1.ExcelExportGroupHeader; } });
var KendoOoxml = require("@progress/kendo-ooxml");
exports.KendoOoxml = KendoOoxml;
__exportStar(require("./ooxml/workbook"), exports);
