// @progress/kendo-react-excel-export/dist/npm/ooxml/workbook.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isWorkbookOptions = exports.toDataURL = exports.workbookOptions = void 0;
var kendo_ooxml_1 = require("@progress/kendo-ooxml");
var ExporterColumns_1 = require("./ExporterColumns");
var getDepth = function (data) {
    return Math.max.apply(Math, data.map(function (i) { return i.level; })) + 1;
};
/**
 * @hidden
 */
var workbookOptions = function (options) {
    var columns = (0, ExporterColumns_1.toExporterColumns)(options.columns);
    var exporter = new kendo_ooxml_1.ExcelExporter({
        columns: columns,
        data: options.data,
        filterable: options.filterable,
        groups: options.group,
        paddingCellOptions: options.paddingCellOptions,
        headerPaddingCellOptions: options.headerPaddingCellOptions,
        hierarchy: options.hierarchy ? { depth: getDepth(options.data), itemLevel: function (item) { return item.level; } } : null,
        collapsible: options.collapsible
    });
    var result = exporter.workbook();
    result.creator = options.creator;
    result.date = options.date;
    result.rtl = options.dir === 'rtl';
    return result;
};
exports.workbookOptions = workbookOptions;
/**
 * @hidden
 */
var toDataURL = function (options) {
    var workbook = new kendo_ooxml_1.Workbook(options);
    return workbook.toDataURL();
};
exports.toDataURL = toDataURL;
/**
 * @hidden
 */
var isWorkbookOptions = function (value) {
    return value && value.sheets;
};
exports.isWorkbookOptions = isWorkbookOptions;
